tests/llm/providers/test_reasoning.py

```
@@ -158,6 +165,19 @@ def test_kimi_code_validation_rejects_unsupported_anthropic_levels() -> None:
             )
 
 
+def test_kimi_code_validation_rejects_budget_tokens_without_budget_level() -> (
+    None
+):
+    with pytest.raises(ValueError, match="budget_tokens.*thinking_level"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)